alphabet.py

```diff
@@ -0,0 +1,49 @@
+from graph import Graph
+
+
+class Alphabet:
+    def __init__(self):
+        """
+        Letters - set of letters encountered so far
+        Relations - graph of relations between letters
+        """
+
+        self.letters = set()
+        self.relations = Graph()
+
+    def _compare_words(self, next_word, prev_word):
+        """
+        Comparing words in pairs in order to find the differing letters and add them to the graph
+        :param string next_word:
+        :param string prev_word:
+        :return:
+        """
+
+        if len(prev_word) == 0:
+            for letter in next_word:
+                self.relations.add_to_graph(None, letter)
+            return
+
+        for i in range(min(len(next_word), len(prev_word))):
```

Comment:
Python has a built-in called `zip` that does just what you need here:

https://docs.python.org/3/library/functions.html#zip